api/envoy/extensions/access_loggers/fluentd/v3/fluentd.proto

```diff
@@ -22,8 +22,20 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 // the Fluentd Forward Protocol as described in: `Fluentd Forward Protocol Specification
 // <https://github.com/fluent/fluentd/wiki/Forward-Protocol-Specification-v1>`_.
 // [#extension: envoy.access_loggers.fluentd]
-// [#next-free-field: 7]
+// [#next-free-field: 8]
 message FluentdAccessLogConfig {
+  message RetryOptions {
+    // The number of times the logger will attempt to connect to the upstream.
+    // The default value is 1, i.e. only a single connection attempt will be made and no reconnection
+    // attempts will be made if the initial connection had failed.
+    // Note: Reconnection attempts will also be made if the initial connection was established, and
+    // disconnected after some time.
```

Comment:
I don't quite understand this. Does this mean that without this setting, if the first connect fails, no logs will ever be sent? When does reconnect happen currently? Can you clarify?

Reply:
Yeah, currently if upstream connection fails/disconnects, logs are lost.
The reason this option got a separate PR was because the underlying AsyncTcpClient had a bug performing reconnects (this was fixed in another PR already)

Reply:
Stepping back I don't understand this tbh. Why would anyone not want to just keep trying to reconnect? It doesn't make sense to me. It seems like the only option we actually want is the min reconnect interval to avoid spin loops?

/wait

Reply:
Actually, I think if we have a reasonable reconnection interval, then the forever reconnecting won't bring overhead and could simplify the code.

So, I agree with matt, interval is much important and useful than this max attempts.

Reply:
What if forever the reconnection will fail? Logs will just keep piling up and bloat. I agree the importance of being able to set backoff interval, and I'll get to work on it as well, but I think that these are two orthogonal problems

Reply:
I think the way this needs to work is:
- There is always a max buffer in place
- Always reconnect with some min reconnect interval

Let's please make it work like this?

Reply:
If the buffer is full while attempting to reconnect, do we throw away new logs?

Reply:
Yes. See max buffer setting above.